### PR TITLE
🔥 feat(codemod): migrate to unify @radix-ui/react-* imports to radix-ui…

### PR DESCRIPTION
**Pull request by nrjdalal**
migrate to unify @radix-ui/react-* imports to radix-ui… in less than a sec via

```sh
npx shadcn migrate radix
```

@shadcn can you release a test package for this? hate waiting for next release when move to radix-ui is planned, early adopters can use the migration easily

you can test the codemod via [smart-registry](https://github.com/nrjdalal/smart-registry) too -

```sh
npx smart-registry@latest --codemod-radix
```

for just apps/v4, use the `--cwd` flag -

```sh
npx smart-registry@latest --codemod-radix --cwd "apps/v4"
```

### CODE REVIEW

File: packages/shadcn/src/migrations/migrate-radix.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  migrationSpinner.text = "Replacing imports in files."
+
+  const unusedPackages = new Set<string>()
+
+  const updateFiles = files.map(async (filepath) => {
+    const absoluteFilepath = path.resolve(uiPath, filepath)
+    const fileContent = await fs.promises.readFile(absoluteFilepath, "utf-8")
+    let transformedContent = fileContent
+      .replace(
+        /import \* as (\w+Primitive) from "@radix-ui\/react-([\w-]+)"/g,
```
Comment (fz6m) [inline packages/shadcn/src/migrations/migrate-radix.ts:34]: Considering different code styles, i suggest using a more general parser like rollup `parseAst` (based on Rust), `babel`, `@typescript-eslint/parser` , or `es-module-lexer` (or `rs-module-lexer`).

Comment (nrjdalal) [reply packages/shadcn/src/migrations/migrate-radix.ts:34]: will look into it for sure

**Comment by nrjdalal** [comment]
Screenshot 2025-06-10 at 1 58 07 AM

needs plus 12 chars -> `"` to `['"]` before integrating

**Comment by shadcn** [comment]
@nrjdalal This looks good. I have been looking into this as well and have a version with ts-morph and package tracking. Would it be okay if I push to your PR?

**Comment by nrjdalal** [comment]
yeah no prob, push it

**Comment by shadcn** [comment]
@nrjdalal I made some changes to handle more edge cases with tests: https://github.com/shadcn-ui/ui/pull/7586
